src/libcore/num/flt2dec/mod.rs

```diff
@@ -425,26 +488,40 @@ pub fn to_shortest_str<'a, T, F>(mut format_shortest: F, v: T,
     match full_decoded {
         FullDecoded::Nan => {
             parts[0] = Part::Copy(b"NaN");
-            Formatted { sign: sign, parts: &parts[..1] }
```

Comment:
Right, separate issue, but what do you think about increasing that threshold (or otherwise altering the heuristic)?